app/javascript/packages/verify-flow/steps/personal-key/download-button.tsx

```diff
@@ -0,0 +1,41 @@
+import type { MouseEventHandler } from 'react';
+import { Button } from '@18f/identity-components';
+import type { ButtonProps } from '@18f/identity-components';
+
+interface DownloadButtonProps {
+  /**
+   * Content of the downloaded file.
+   */
+  content: string;
+
+  /**
+   * File name to use for downloaded file.
+   */
+  fileName: string;
+}
+
+function DownloadButton({ content, fileName, ...buttonProps }: DownloadButtonProps & ButtonProps) {
+  const download: MouseEventHandler = (event) => {
+    buttonProps.onClick?.(event);
+
+    if ('msSaveBlob' in window.navigator) {
+      event.preventDefault();
+
+      const filename = (event.target as HTMLButtonElement).getAttribute('download');
+      const blob = new Blob([content], { type: 'text/plain' });
+
+      (window.navigator as any).msSaveBlob(blob, filename);
```

Comment:
since we do the `if ... in` check, what if we cast as some sort of like `window | ProprietaryBrowser` or something so we don't have to cast as `any` again?

Or do a `isProprietaryBrowser(w: window | ProprietyaryBrowser): is ProprietaryBrowser`

Reply:
> since we do the `if ... in` check, what if we cast as some sort of like `window | ProprietaryBrowser` or something so we don't have to cast as `any` again?
> 
> Or do a `isProprietaryBrowser(w: window | ProprietyaryBrowser): is ProprietaryBrowser`

Updated in b333339ed. Type narrowing is pretty neat!

Reply:
big win! love it